src/m365/planner/commands/plan/plan-details-get.ts

```diff
@@ -0,0 +1,177 @@
+import { Logger } from '../../../../cli';
+import { CommandOption} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  planId?: string;
+  planTitle?: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanDetailsGetCommand extends GraphItemsListCommand<any> {
+  private groupId: string = "";
+
+  public get name(): string {
+    return commands.PLAN_DETAILS_GET;
+  }
+
+  public get description(): string {
+    return 'Get details of a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planTitle = typeof args.options.planTitle !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.planId) {
+      this
+        .getPlanDetails(args)
+        .then((res: any): void => {
+          logger.log(res);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
```

Comment:
To avoid repeating this code, let's centralize how we retrieve plan details so that we only need to output it once